server/src/main/java/org/elasticsearch/search/query/QueryPhase.java

```diff
@@ -300,12 +297,6 @@ static boolean executeInternal(SearchContext searchContext) throws QueryPhaseExe
                 restoreTopFieldDocs(queryResult, sortAndFormatsForRewrittenNumericSort);
             }
 
-            ExecutorService executor = searchContext.indexShard().getThreadPool().executor(ThreadPool.Names.SEARCH);
-            if (executor instanceof QueueResizingEsThreadPoolExecutor) {
-                QueueResizingEsThreadPoolExecutor rExecutor = (QueueResizingEsThreadPoolExecutor) executor;
-                queryResult.nodeQueueSize(rExecutor.getCurrentQueueSize());
-                queryResult.serviceTimeEWMA((long) rExecutor.getTaskExecutionEWMA());
-            }
```

Comment:
These stats are still used for adaptive replica selection. This is what I meant when we discussed this about the `QueueResizingEsThreadPoolExecutor` working for both the little's law part (the part we wanted to remove) as well as for adaptive replica selection.

Reply:
You're absolutely right. I was too eager to delete code here. Will be interesting to see if any test catches this. Will rework this PR tomorrow.

Reply:
I've addressed this now in 382d40b